repology/fetchers/fetchers/repodata.py

```diff
@@ -25,21 +25,28 @@
 
 
 class RepodataFetcher(ScratchFileFetcher):
-    def __init__(self, url: str, fetch_timeout: int = 60):
+    def __init__(self, url: str, fetch_timeout: int = 60, use_db: bool = False):
         super(RepodataFetcher, self).__init__(binary=True)
 
         self.url = url
         self.fetch_timeout = fetch_timeout
+        self.use_db = use_db
```

Comment:
I'd instead set `primary_key = 'primary'` as the class member. Derived class would then override it to `primary_db`. Constructor argument goes away (which is good as it would allow confusing uses), and you don't even need to override constructor in the derived class.

here and remove the constructor argument (which may lead to confusing uses). `RepodataDbFetcher` constructor would set it to `primary_db`

Reply:
Great suggestions, this definitely cleans it up.  Fixing in v2 ...